Rename commands for org-roam v2 compatibility

kappa opened the pull request:
Fixes #14929

The main org-roam command "org-roam-find-file" was replaced with "org-roam-node-find". I decided to leave 
the old key sequence "a o r f" as an alternative for the new command alongside "a o r n" so that people who don't care about the internal naming of org-roam objects could keep using their muscle memory.

snuffop commented:
:+1:

emacs18 commented:
`org-roam-tag-delete` also seems to have been removed.
I see `org-roam-tag-remove` instead.

emacs18 commented:
`READEME.org` should also be updated, i.e., the key bindings at the end of the file.

Claude-Ray commented:
The same goes for `org-roam-dailies-find-*`. They are warned as obsolete since `org-roam-dailies-goto-*` have been created.

emacs18 commented:
It seems like key bindings on lnes 960 and 961 are incorrect, i.e.,
```diff
+        "rtr" 'org-roam-tag-add
+        "rtd" 'org-roam-tag-remove
```
In global key bindings, the same key commands are bound to "aorta" and "aortr".

emacs18 commented:
Should `org-roam-setup` function be called to set things up?

magthe commented:
> Should `org-roam-setup` function be called to set things up?

The docs, https://www.orgroam.com/manual.html#Setting-up-Org_002droam, suggests it should be called, yes.

ZacheryPD commented:
Feel free to disregard, but would it also be a good idea to add a keybinding for org-roam-capture?

jason--liu commented:
Why hasn't this patch been merged yet? ^^

kappa commented:
> Feel free to disregard, but would it also be a good idea to add a keybinding for org-roam-capture?

Are there scenarios where you want org-roam-capture outside of adding a new node? And that one already has 2 keybindings -- "SPC a o r f/n".

duianto commented:
There are two org-roam fix PR's:
Rename commands for org-roam v2 compatibility #14930
Change org-roam keybindings to use new v2 format #14939

We should combine the changes into one.

A quick glance shows these differences.
- two node find key bindings in the first PR
```
        "aorf" 'org-roam-node-find
        "aorn" 'org-roam-node-find
```
the second PR only has the first.
- tag removal has two different key bindings
First PR:
```
"aortr" 'org-roam-tag-remove
```
Second PR:
```
"aortd" 'org-roam-tag-remove
```
- the function org/init-org-roam-server is removed in the second PR
```
(defun org/init-org-roam-server()
  (use-package org-roam-server
    :defer t
    :init
    (progn
      (spacemacs/set-leader-keys "aors" 'org-roam-server-mode)
      (spacemacs/set-leader-keys-for-major-mode 'org-mode "rs" 'org-roam-server-mode))))
```

KjartanOli commented:
> > Feel free to disregard, but would it also be a good idea to add a keybinding for org-roam-capture?
> 
> Are there scenarios where you want org-roam-capture outside of adding a new node? And that one already has 2 keybindings -- "SPC a o r f/n".

If I remember correctly there is a separate function `org-roam-capture` which behaves slightly differently from `org-roam-node-find`.

duianto commented:
Let's use this PR since it was opened first.

duianto commented:
@kappa can you add an entry in
https://github.com/syl20bnr/spacemacs/blob/develop/CHANGELOG.develop

That mentions both of you org-roam fix PR authors. Something like this:
```
- Fix org-roam v2 compatibility (thanks to Alex Kapranoff and Caleb Rogers)
```
at the end of the org section in:
`*** Layer changes and fixes`
`**** Org`
on line 3042, above the heading: `**** Osx`

Note, there's also a Org section in `*** Breaking Changes`, don't add it there.

ZacheryPD commented:
> > > Feel free to disregard, but would it also be a good idea to add a keybinding for org-roam-capture?
> 
> > 
> 
> > Are there scenarios where you want org-roam-capture outside of adding a new node? And that one already has 2 keybindings -- "SPC a o r f/n".
> 
> 
> 
> If I remember correctly there is a separate function `org-roam-capture` which behaves slightly differently from `org-roam-node-find`. 

This is correct. I'm new to roam so I may be mistaken but if I recall correctly capture can have templates like org-capture.

duianto commented:
@kappa After deciding if any of the other PR's changes should be incorporated.
Change org-roam keybindings to use new v2 format #14939

And if there are other changes that should be made.

Could you squash the commits into 1 when this is ready to review.
The recommendation is to keep simple PRs to one commit per PR.
https://github.com/syl20bnr/spacemacs/blob/develop/CONTRIBUTING.org#pull-request

kappa commented:
@duianto -- great instructions on squashing using magit, thank you!

I incorporated differences from @komali2's PR and added a new keybinding for org-roam-capture as suggested by @ZacheryPD.

duianto commented:
org/readme changes:
- Add an entry for the org-roam-capture key binding
- Remove the org-roam server key binding, since the package isn't used at the moment.
- Align the key binding table.
With the cursor in the table, press `TAB` to call `org-cycle`. It's docstring says:
>When inside a table, re-align the table and move to the next field.

duianto commented:
Thank you for contributing to Spacemacs.

The readme has been updated.
[docs] Update org-roam key bindings https://github.com/syl20bnr/spacemacs/commit/3b8bb7f5b7305a04438d609036461f3ddbe1e6cd

The org switch buffer key binding was also removed.
And the alias key binding was moved up,
to group it with the other key bindings without a prefix.
